go-controller/cmd/ovnkube-trace/udn.go

```diff
@@ -41,28 +44,31 @@ func findUserDefinedNetworkVRFTableIDs(coreclient *corev1client.CoreV1Client, re
 	return string(nodesTableIDsJSON), nil
 }
 
-func findUserDefinedNetworkVRFTableID(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, node *corev1.Node, ovnNamespace, networkName string) (uint, error) {
+func findUserDefinedNetworkVRFTableID(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, node *corev1.Node, ovnNamespace, networkName string) (*uint, error) {
 	ovnKubePodName, err := getOvnKubePodOnNode(coreclient, ovnNamespace, node.Name)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	mgmtPortLinkName := util.GetSecondaryNetworkPrefix(networkName) + types.K8sMgmtIntfName
-	ipLinkCmd := "ip -j link show dev " + mgmtPortLinkName
-	stdout, stderr, err := execInPod(coreclient, restconfig, ovnNamespace, ovnKubePodName, ovnKubeNodePodContainers[1], ipLinkCmd, "")
+	ipLinksCmd := "ip -j link"
+	stdout, stderr, err := execInPod(coreclient, restconfig, ovnNamespace, ovnKubePodName, ovnKubeNodePodContainers[1], ipLinksCmd, "")
 	if err != nil {
-		return 0, fmt.Errorf("%s: %s: %w", stdout, stderr, err)
+		return nil, fmt.Errorf("%s: %s: %w", stdout, stderr, err)
 	}
 	links := []struct {
-		Index uint `json:"ifindex"`
+		Name  string `json:"ifname"`
+		Index uint   `json:"ifindex"`
 	}{}
 	if err := json.Unmarshal([]byte(stdout), &links); err != nil {
-		return 0, err
+		return nil, err
 	}
-	if len(links) < 1 {
-		return 0, fmt.Errorf("link '%s' not found", mgmtPortLinkName)
+	mgmtPortLinkName := util.GetSecondaryNetworkPrefix(networkName) + types.K8sMgmtIntfName
```

Comment:
Ok, but we are going to need the networkID instead of name so we don't reach the linux device name limit.

Reply:
ack then all the more need for a util func given this would break otherwise
as spoken offline let's add a trim and some details on the util functional on the limit being 15chars